Bottom: 54969dd5cadb4809451a5469fc6fd15e6bf7e1ff
Top:    b91c156d9e5c0d83c1e46e8d8e1677f305b7083d
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-18 16:36:07 -0700

rsockets: Define options specific to rsockets

Allow a user to control some of the RDMA related attributes
of an rsocket through setsockopt/getsockopt.  A user specifies
that the rsocket should be modified through SOL_RDMA level.

This patch provides the initial framework.  Subsequent patches
will add the configurable parameters.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/include/rdma/rsocket.h b/include/rdma/rsocket.h
index 5310d33..87ee943 100644
--- a/include/rdma/rsocket.h
+++ b/include/rdma/rsocket.h
@@ -72,6 +72,8 @@ int rselect(int nfds, fd_set *readfds, fd_set *writefds,
 int rgetpeername(int socket, struct sockaddr *addr, socklen_t *addrlen);
 int rgetsockname(int socket, struct sockaddr *addr, socklen_t *addrlen);
 
+#define SOL_RDMA 0x10000
+
 int rsetsockopt(int socket, int level, int optname,
 		const void *optval, socklen_t optlen);
 int rgetsockopt(int socket, int level, int optname,
diff --git a/src/rsocket.c b/src/rsocket.c
index 59cc13c..ee31e0c 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1710,6 +1710,8 @@ int rsetsockopt(int socket, int level, int optname,
 			break;
 		}
 		break;
+	case SOL_RDMA:
+		break;
 	default:
 		break;
 	}
@@ -1775,6 +1777,9 @@ int rgetsockopt(int socket, int level, int optname,
 			break;
 		}
 		break;
+	case SOL_RDMA:
+		ret = ENOTSUP;
+		break;
 	default:
 		ret = ENOTSUP;
 		break;
